client/app/components/land-use-action.js

```diff
@@ -61,64 +61,65 @@ export const allLandUseActionOptions = [
-  // existing in CRM ("existing")
-  get landUseActionSelections() {
-    const { pasForm } = this.args;
-    // Filter the "new" actions array so that the action does not exist on BOTH the "exisiting" and "new" arrays,
-    // when user enters a value in the action inputs (and therefore updates the model)
-    const filteredNewActions = this.newLandUseActionSelections.filter((action) => !pasForm[action.countField] && !pasForm[action.attr1] && !pasForm[action.attr2]);
-    return [
-      ...this.existingLandUseActionSelections,
-      ...filteredNewActions,
-    ];
-  }
+  // sorting: actions from db all on bottom in alphabetical order
+  // sorting: new actions all on top, most recently added first
+  get sortedSelectedActions() {
+    const newActions = this.actionsAddedByUser;
+
+    return this.selectedActions.sort(function (a, b) {
+      const includesA = newActions.includes(a);
+      const includesB = newActions.includes(b);
 
-  get availableLandUseActionOptions() {
-    return allLandUseActionOptions.filter((option) => !this.landUseActionSelections.includes(option));
+      // if actions are both new -- sort most recently added on top
+      if (includesA && includesB) {
+        return newActions.indexOf(a) - newActions.indexOf(b);
+      // if actions are both from db -- sort alphabetically
+      } else if (!includesA && !includesB) { // eslint-disable-line
+        return (a.name > b.name) ? 1 : -1;
+      // if top action is new and bottom action is from db -- don't reorder
+      } else if (includesA && !includesB) {
+        return -1;
+      // if top action is from db and bottom action is new -- reorder
+      } else if (!includesA && includesB) {
+        return 1;
+      } return 0;
+    });
   }
 
-  get sortedAvailableLandUseActionOptions() {
-    return this.availableLandUseActionOptions.sort((a, b) => ((a.name > b.name) ? 1 : -1));
+  // actions that appear in dropdown sorted alphabetically
+  get availableActions() {
+    const availableLandUseActions = allLandUseActionOptions.filter((option) => !this.selectedActions.includes(option));
+    return availableLandUseActions.sort((a, b) => ((a.name > b.name) ? 1 : -1));
   }
 
-  // when a user selects an option from the dropdown menu,
-  // the option is filtered out of the availableLandUseActionOptions
-  // (and therefore removed from the dropdown).
-  // The selected option is added to the newLandUseActionSelections array
-  // (and therefore displays in the list of selected actions).
   @action
-  addNewLandUseActionSelection() {
-    const selection = this.selectedLandUseAction.countField;
+  addSelectedAction() {
+    this.args.pasForm[this.selectedAction.countField] = 1;
+    this.actionsAddedByUser.unshiftObject(this.selectedAction);
 
-    if (selection) {
-      this.newLandUseActionSelections.pushObject(this.selectedLandUseAction);
-      this.selectedLandUseAction = {};
-    }
+    this.selectedAction = {};
```

Comment:
What's another way of handling selectedAction? What would the code look like if you had the dropdown call `addSelectedAction` directly? I'm wondering — how can we make it to where we don't need to store a reference to the "selectedAction", any state at all.

Any time you store a reference on the class (setting `selectedAction` or something), you have to manage it through setting it back, etc.